tests/session_test.go

```diff
@@ -324,3 +324,35 @@ func parseDateFromDelimiter(messageDelimiter string) (t time.Time, err error) {
 
 	return time.Parse("Mon Jan _2 15:04:05 2006", strings.TrimSpace(strings.Join(split[2:], " ")))
 }
+
+func TestTooManyInvalidCommand(t *testing.T) {
+	runOneToOneTestWithAuth(t, defaultServerOptions(t), func(c *testConnection, s *testSession) {
+		for i := 1; i <= 19; i++ {
+			c.C(fmt.Sprintf("%d FOO", i))
+			c.Sx(fmt.Sprintf("%d BAD mismatched input", i))
+		}
+		c.C(fmt.Sprintf("100 FOO"))
+		c.S(fmt.Sprintf("100 BAD too many invalid IMAP command"))
+		// expect server to trigger the error as well
+		require.Errorf(t, <-s.server.GetErrorCh(), "100 BAD too many invalid IMAP command")
+	})
+}
+
+func TestResetTooManyInvalidCommand(t *testing.T) {
+	runOneToOneTestWithAuth(t, defaultServerOptions(t), func(c *testConnection, s *testSession) {
+		for i := 1; i <= 19; i++ {
+			c.C(fmt.Sprintf("%d FOO", i))
+			c.Sx(fmt.Sprintf("%d BAD mismatched input", i))
+		}
+		c.C(fmt.Sprintf("100 NOOP"))
+		c.Sx(fmt.Sprintf("100 OK"))
+		for i := 1; i <= 19; i++ {
+			c.C(fmt.Sprintf("%d FOO", i))
+			c.Sx(fmt.Sprintf("%d BAD mismatched input", i))
+		}
+		c.C(fmt.Sprintf("100 FOO"))
+		c.S(fmt.Sprintf("100 BAD too many invalid IMAP command"))
```

Comment:
```suggestion
		c.C("100 FOO").BAD("100")
```
you can do this too (if you don't care about the exact error string, which maybe you do)